### PR TITLE
Implement placement-in protocol for `VecDeque`

### PR DESCRIPTION
**Pull request by F001**
CC #30172

r? @nagisa

### CODE REVIEW

**Comment by nagisa** [comment]
Looks great to me.

**Comment by nagisa** [comment]
@bors r+

Thanks!

File: src/libcollections/vec_deque.rs [hunk 1]
```diff
@@ -2442,6 +2484,98 @@ impl<T> From<VecDeque<T>> for Vec<T> {
...
+#[derive(Debug)]
+pub struct PlaceBack<'a, T: 'a> {
+    vec_deque: &'a mut VecDeque<T>,
+}
+
+#[unstable(feature = "collection_placement",
+           reason = "placement protocol is subject to change",
+           issue = "30172")]
+impl<'a, T> Placer<T> for PlaceBack<'a, T> {
+    type Place = PlaceBack<'a, T>;
```
Comment (tikue) [inline src/libcollections/vec_deque.rs]: Could this be `Self`?

Comment (F001) [inline src/libcollections/vec_deque.rs]: Thanks. It's the same. And I don't bias towards any one.

**Pull request merged**